docs/index.md

```diff
@@ -56,77 +60,125 @@ $> python -c "import luxos; print(luxos.__version__, luxos.__hash__)"
+- [luxos.util.launch](luxos.utils.launch) - run a command on multiple miners
+
+
+#### the rexec function
+The `rexec` function allows to send a command to a miner and return the response:
 ```python
-   >>> import asyncio
-   >>> from luxos import utils
-   >>> asyncio.run(utils.rexec("127.0.0.1", 4028, "vesion"))
-   {'STATUS': [{'Code': 22, 'Description': 'LUXminer ...
+   
+    from luxos import utils
+    res = utils.asyncio.run(utils.rexec("127.0.0.1", 4028, "version"))
+    # this is the validate helper
+    print(utils.validate(res, "VERSION", 1, 1))
 ```
-`rexec` takes care of formatting the parameters also, the full signature is:
+
+The full signature for `rexec` takes care of formatting the parameters, the full signature is:
 ```python
 rexec(
     host="127.0.0.1", port=4028, 
     cmd="version", parameters="",
     timeout=2., retry=1, retry_delay=3.)
 ```
-`parameters` can be a string, a list of any type (it will be converted into a str) or a dictionary (same conversion to string will apply).  `timeout` is the timeout for a call `retry` is the number of try before giving up `retry_delay` controls the delay between retry.
+`parameters` can be a string, a list of any type (it will be converted into a str) 
+or a dictionary (same conversion to string will apply).  
+
+`timeout` is the timeout for a call `retry` is the number of try 
+before giving up `retry_delay` controls the delay between retry.
+
+> **NOTE** `rexec` is an async function, 
+> but there's a sync version under `luxos.syncops.rexec`.
 
 #### launch
 The `luxos.utils.lauch` allows to rexec commands to a list of miners stored in a csv file.
 This all-in-one function, allows batched operations on a set of miners, taking care of all details.
 
 ```python
-   >>> import asyncio
-   >>> from luxos import utils
-
-   # load miners addresses from miners.csv
-   >>> miners = utils.load_ips_from_csv("miners.csv")
+import asyncio
+from luxos import utils
 
+# load miners addresses from miners.csv
+addresses = utils.load_ips_from_csv("miners.csv")
 
-   # define a task with (host, port) signature, acting on a miner
-   >>> async def task(host: str, port: int):
-   ...   return await utils.rexec(host, port, "version")
 
+# define a task with (host, port) signature, acting on a miner
+async def task(host: str, port: int):
+    return await utils.rexec(host, port, "version")
 
-   # execute task across all miners, batched will limit execution rate
-   >>> asyncio.run(utils.launch(addresses, task, batch=None))
-   [{'STATUS': [{'Code': 22, 'Description': 'LUXmin ....
+# execute task across all miners, batch will limit execution rate to 4
+asyncio.run(utils.launch(addresses, task, batch=4))
 
-   # the one-liner for the version command
-   >>> asyncio.run(utils.launch(addresses, utils.rexec, "version", batch=None))
+# a one liner
+asyncio.run(utils.launch(addresses, lambda h, p: utils.rexec(h, p, "version"), batch=None))
```

Comment:
version command doesn't have params

Reply:
the `batch=None` is not an argument to version command, is an argument to the enclosing launch. Or do I miss something else?

Reply:
I meant that it would be nice to have an example of launching a command that requires params.